### PR TITLE
Close dig objects on Reset()

### PR DESCRIPTION
**Pull request by glibsm**
GFM-377

### CODE REVIEW

File: dig/graph.go [hunk 1]
```diff
@@ -133,6 +134,16 @@ func (g *graph) Reset() {
 	g.Lock()
 	defer g.Unlock()
 
+	for _, node := range g.nodes {
+		if v, err := node.value(g); err == nil {
+			if nc, ok := v.Interface().(io.Closer); ok {
+				// Error is ignored. What else can be done here, should Reset() return an error?
+				// Assignment is to satisfy errcheck :(
+				err = nc.Close()
```
Comment (sectioneight) [inline dig/graph.go]: log it?

File: dig/graph.go [hunk 2]
```diff
@@ -128,11 +129,20 @@ func (g *graph) RegisterAll(cs ...interface{}) error {
 	return nil
 }
 
-// Reset the graph by removing all the registered nodes
 func (g *graph) Reset() {
 	g.Lock()
 	defer g.Unlock()
 
+	for _, node := range g.nodes {
+		if v, err := node.value(g); err == nil {
```
Comment (alsamylkin) [inline dig/graph.go]: v can be nil, right?

Comment (alsamylkin) [inline dig/graph.go]: Oh, it can't

File: dig/graph.go [hunk 3]
```diff
@@ -128,11 +129,20 @@ func (g *graph) RegisterAll(cs ...interface{}) error {
 	return nil
 }
 
-// Reset the graph by removing all the registered nodes
```
Comment (alsamylkin) [inline dig/graph.go]: update, not remove?

Comment (glibsm) [inline dig/graph.go]: These need to be removed, because they appear in there different places. And as abg pointed out, these don't make it into GoDoc

**Comment by glibsm** [comment]
going to table this for now

**Pull request closed without merging**